src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -163,8 +166,13 @@ public int numberOfShards() {
     }
 
     @Override
-    public UnmodifiableIterator<IndexShard> iterator() {
-        return shards.values().iterator();
+    public Iterator<IndexShard> iterator() {
```

Comment:
I disagree - I removed this weirdness here since we wanna code against interfaces here